### ISSUE TITLE
Implicit resolution problem for `Transformer[Option[A], B]`

### ISSUE THREAD

**Issue by bjhartin** [scalalandio/chimney#113]
I'm probably missing something here.

This has been discussed in #62 - mapping from options to required fields, throwing errors on `None`.  The approach there works, _if_ mapping from `Option[A]` to `A`.  However, mapping from `Option[A]` to `B` fails in some cases, _even if_ a `Transformer[A, B]` is in scope.

Example:

```scala
object ChimneyTest extends App {
  import io.scalaland.chimney.Transformer
  import io.scalaland.chimney.dsl._

  case class Bar1(i: Int) // 'extends AnyVal' makes no difference
  case class Bar2(i: String) // 'extends AnyVal' makes no difference
  case class Bar3(i: Option[Int])

  case class Baz1(b: Bar1)
  case class Baz2(b: Option[Bar1])
  case class Baz3(b: Bar2)
  case class Baz4(b: Option[Bar2])

  implicit val intToString: Transformer[Int, String] = _.toString

  implicit def idTransform[A]: Transformer[A, A] = identity[A]

  // Throws on `None` - not ideal but Chimney doesn't represent failure, e.g. `Either[Error, B]`
  implicit def transformFromOptional[A, B](implicit ev: Transformer[A, B]): Transformer[Option[A], B] =
    (a: Option[A]) => a.map(ev.transform).getOrElse(throw new NoSuchElementException("Optional field was none"))

  // (Failing example first, then simpler, passing examples)

  // FAILS TO COMPILE:
  // Baz2(Option(Bar1(1))) => Baz3(Bar2("1")) fails
  // Should be available through transformFromOptional, case class support and intToString
  //assert(Baz2(Option(Bar1(1))).transformInto[Baz3] == Baz3(Bar2("1")))

  assert(1.transformInto[String] == "1")

  assert(Option(1).transformInto[Int] == 1)

  assert(Option(1).transformInto[String] == "1")

  assert(Bar1(1).transformInto[Bar2] == Bar2("1"))

  assert(Option(Bar1(1)).transformInto[Bar1] == Bar1(1))

  assert(Baz2(Option(Bar1(1))).transformInto[Baz1] == Baz1(Bar1(1)))

  assert(Option(Bar1(1)).transformInto[Bar2] == Bar2("1"))

  assert(Baz2(Option(Bar1(1))).transformInto[Baz4] == Baz4(Option(Bar2("1"))))

  assert(Bar3(Option(1)).transformInto[Bar2] == Bar2("1"))
}
```

The failure is:

```
Error:(25, 43) Chimney can't derive transformation from Baz2 to Baz3
Bar2
  i: java.lang.String - no accessor named i in source type scala.Option
Consult https://scalalandio.github.io/chimney for usage examples.
  assert(Baz2(Some(Bar1(1))).transformInto[Baz3] == Baz3(Bar2("1")))
```

**Comment by jchapuis** [comment]
I'm running into the same difficulties here, trying to achieve boilerplate-free proto3 conversion

**Comment by krzemin** [comment]
I guess that after #116 this is no longer relevant. Please re-open if I missed something.
